internal/controllers/clusterextension_controller_test.go

```diff
@@ -229,22 +245,27 @@ func TestClusterExtensionChannelVersionExists(t *testing.T) {
 	require.Equal(t, metav1.ConditionTrue, cond.Status)
 	require.Equal(t, ocv1alpha1.ReasonSuccess, cond.Reason)
 	require.Equal(t, "resolved to \"quay.io/operatorhubio/prometheus@fake1.0.0\"", cond.Message)
-	cond = apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeInstalled)
-	require.NotNil(t, cond)
-	require.Equal(t, metav1.ConditionUnknown, cond.Status)
-	require.Equal(t, ocv1alpha1.ReasonInstallationStatusUnknown, cond.Reason)
-	require.Equal(t, "bundledeployment status is unknown", cond.Message)
-
-	t.Log("By fetching the bundled deployment")
-	bd := &rukpakv1alpha2.BundleDeployment{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{Name: extKey.Name}, bd))
-	require.Equal(t, "core-rukpak-io-registry", bd.Spec.ProvisionerClassName)
-	require.Equal(t, installNamespace, bd.Spec.InstallNamespace)
-	require.Equal(t, rukpakv1alpha2.SourceTypeImage, bd.Spec.Source.Type)
-	require.NotNil(t, bd.Spec.Source.Image)
-	require.Equal(t, "quay.io/operatorhubio/prometheus@fake1.0.0", bd.Spec.Source.Image.Ref)
 
-	verifyInvariants(ctx, t, reconciler.Client, clusterExtension)
+	// we're skipping based on same loadError since Status.Condition setting path is skipped and bundle deployment is absent
+	// also invariants will be different
+	if err != nil && err.Error() != utilerrors.NewAggregate([]error{loadError}).Error() {
+		cond = apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeInstalled)
```

Comment:
Do we need to nail down that these conditions have been set?

Reply:
same comment as below: @joelanford wdyt? there is a lot more mocking to do if we need to reach the code that sets these exactly as the current test(s) want.

Reply:
IMO, we should be able to only modify the behavior associated with the removal of the `BundleDeployment` API and the status condition that should be present on the `ClusterExtension` API. Other than those changes I would expect that, if our unit tests are well written, the tests should remain as is and function appropriately. 

That being said, it does seem like in order to get our controller logic more easily unit testable it would require a fairly significant refactor of the `reconcile()` function in which more abstractions are created to mock out and maybe that is better suited as a follow up done ASAP after the initial PR merges. IIRC consensus from the meeting where we walked through the main PR was that e2e tests are a must and should only be skipped with verifiably good reason and unit tests while not ideal to be skipped may be acceptable to un-skipped in a follow up with the abstractions.

In my head the very high-level flow seems to be:
- Resolution to get Bundle for ClusterExtension
- Unpack Bundle image
- Convert bundle contents to Helm Chart
- Apply Helm Chart

I would expect that there is some abstraction for each of these steps that can be mocked, making our controller much easier to unit test

Reply:
+1 to Bryce's comment. And then we would also want unit tests for each of the actual real implementations of those abstractions.

But also, the verifyInvariants test is there to make sure that no matter when we return from `reconcile` we have formed an opinion on every Condition type that we support. It's okay if we don't progress all the way through all the code paths, but this sounds like a good example of this test doing its job. It means that the paths that we _do_ test don't actually make sure that all of the conditions are initialized.

Reply:
> verifyInvariants test is there to make sure that no matter when we return from reconcile we have formed an opinion on every Condition type that we support.

This is a mess given we have 8 condition types. This is fixed for now, but there is a follow up created to re-evaluate the condition types. Resolving this conversation for now.